postgresql/postgresql_plugin.cc

```diff
@@ -0,0 +1,832 @@
+/* postgresql_plugin.cc
+   Mathieu Marquis Bolduc, 16 July 2016
+   Copyright (c) 2015 Datacratic Inc.  All rights reserved.
+*/
+
+#include "mldb/core/plugin.h"
+#include "mldb/core/procedure.h"
+#include "mldb/core/function.h"
+#include "mldb/core/dataset.h"
+#include "mldb/soa/credentials/credentials.h"
+#include "mldb/types/structure_description.h"
+#include "mldb/types/any_impl.h"
+
+#include <postgresql/libpq-fe.h>
+
+#include <unordered_set>
+
+using namespace std;
+
+Datacratic::MLDB::Plugin *
+mldbPluginEnterV100(Datacratic::MLDB::MldbServer * server)
+{
+    return nullptr;
+}
+
+namespace Datacratic {
+namespace MLDB {
+
+const Package & postgresqlPackage()
+{
+    static const Package result("postgresql");
+    return result;
+}
+
+//#define POSTGRESQL_VERBOSE(x) x
+#define POSTGRESQL_VERBOSE(x)
+
+/*****************************************************************************/
+/* POSTGRESQL UTILS                                                          */
+/*****************************************************************************/
+namespace {
+CellValue getCellValueFromPostgres(const PGresult *res, int i, int j)
+{
+    int postgrestype = PQftype(res, j);
+    if (postgrestype == 23) {
+        return CellValue(atoi(PQgetvalue(res, i, j)));
+    }
+    else if (postgrestype == 701) {
+        return CellValue(atof(PQgetvalue(res, i, j)));
+    }
+    else {
+        return CellValue(PQgetvalue(res, i, j));
+    }
+}
+
+pg_conn* startPostgresqlConnection(string databaseName, string host, int port)
+{
+    auto creds = getCredential("postgresql", host);
+
+    string connString;
+    connString += "dbname=" + databaseName
+               + " host=" + host
+               + " port=" + std::to_string(port)
+               + " user=" + creds.id
+               + " password=" + creds.secret;
+
+    POSTGRESQL_VERBOSE(cerr << connString << endl;)
+
+    auto conn = PQconnectdb(connString.c_str());
+    if (PQstatus(conn) != CONNECTION_OK)
```

Comment:
Can we be more specific about the error?  It would be particularly helpful to know if it is a connection problem or a credential one.